apiserver/facades/agent/upgrader/unitupgrader.go

```diff
@@ -125,6 +118,82 @@ func (u *UnitUpgraderAPI) DesiredVersion(ctx context.Context, args params.Entiti
 	return params.VersionResults{Results: result}, nil
 }
 
+// SetTools is responsible for updating a a set of entities reported agent
+// version.
+func (u *UnitUpgraderAPI) SetTools(
+	ctx context.Context,
+	args params.EntitiesVersion,
+) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.AgentTools)),
+	}
+
+	for i, entityVersion := range args.AgentTools {
+		tag, err := names.ParseTag(entityVersion.Tag)
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
+			continue
+		}
+
+		if !u.authorizer.AuthOwner(tag) {
```

Comment:
We set up `getCanReadWrite` when registering, implemented as a call to `AuthOwner`,  which we pass to tools getter. We should use that here too instead of `AuthOwner` directly. Strictly speaking, we should just require `canRead` for getting agent binaries.